website/content/docs/nia/terraform-modules.mdx

```diff
@@ -124,6 +124,18 @@ We recommend that if you make a module with with a catalog-services condition, t
 
 Similarly, if you include the `catalog_services` variable in your module, we recommend that you also document this usage in the README. Users of your module will then know to set the catalog-services condition [`source_includes_var`](/docs/nia/configuration#source_includes_var) configuration to be true. When this field is set to true, Consul-Terraform-Sync will declare the `catalog_services` variable in the generated root module, and pass the variable to a child module. Therefore, if this field is configured inconsistently, Consul-Terraform-Sync will error and exit.
 
+### Consul KV Variable
+If you are creating a module for a [consul-kv condition](/docs/nia/tasks#consul-kv-condition), then you have the option to add the `consul_kv` variable, which contains a map of the keys and values for the Consul KV pairs. If your module would benefit from consuming this information, you can copy the `consul_kv` variable declaration to your `variables.tf` file in addition to the other variables.
+
+```hcl
+variable "consul_kv" {
+	description = "Keys and values of the Consul KV pairs monitored by Consul-Terraform-Sync"
+	type = map(string)
+}
+```
+
+If you include the `consul_kv` variable in your module, we recommend that you also document this usage in the README. Users of your module will then know to set the consul-kv condition [`source_includes_var`](/docs/nia/configuration#source_includes_var-1) configuration to be true. When this field is set to true, Consul-Terraform-Sync will declare the `consul_kv` variable in the generated root module, and pass the variable to a child module. Therefore, if this field is configured inconsistently, Consul-Terraform-Sync will error and exit.
```

Comment:
I just want to make sure I'm understanding this correctly.

If a module contains the consul_kv variable, then you  **must** set `source_includes_var` to true in the `condition "schedule"` block?

Reply:
Yeah, in general, if a module declares a variable, then Terraform will fail if that variable is not provided. For CTS, that means the user has to set `source_includes_var` to true so that CTS creates and passes that variable.

Reply:
```suggestion
If your module contains the `consul_kv` variable, we recommend documenting usage in the README file so that users know to set the [`source_includes_var`](/docs/nia/configuration#source_includes_var-1) configuration to `true` in the `consul-kv` condition. Setting the field to `true` instructs Consul-Terraform-Sync to declare the `consul_kv` variable in the generated root module and pass the variable to a child module. Therefore, if this field is configured inconsistently, Consul-Terraform-Sync will error and exit.
```
Just trying some different wording to see if I can make it easier to parse. Please ignore the suggestion if it isn't an improvement.